probe/endpoint/procspy/fixture.go

```diff
@@ -21,10 +13,15 @@ func (f *fixedConnIter) Next() *Connection {
 	return &car
 }
 
-// SetFixtures is used in test scenarios to have known output.
-func SetFixtures(c []Connection) {
-	cbConnections = func(bool, process.Walker) (ConnIter, error) {
-		f := fixedConnIter(c)
-		return &f, nil
-	}
+// FixedScanner implements ConnectionScanner and uses constant Connection and
+// ConnectionProcs.  It's designed to be used in tests.
+type FixedScanner []Connection
+
+// Connections implements ConnectionsScanner.Connections
+func (s FixedScanner) Connections(_ bool) (ConnIter, error) {
+	iter := fixedConnIter(s)
+	return &iter, nil
 }
+
+// Stop implements ConnectionsScanner.Stop (dummy since there is no background work)
+func (s FixedScanner) Stop() {}
```

Comment:
If this is only used in tests let's define it in a test package. Looks like it's only used in `probe/endpoint/reporter_test.go`.